### ISSUE TITLE
Inline syntax highlighting is broken in the latest version(s)

### ISSUE THREAD

**Issue by davidcelis** [voracious/ink-mde#56]
It looks like recent changes to reduce the final package size have omitted whatever code/files were necessary for the inline Markdown syntax highlighting. I recently updated the version of Ink that I use and pointed at `https://cdn.jsdelivr.net/npm/ink-mde@0.20.1/+esm` and just noticed that the inline Markdown styles no longer work. I had to go back a few versions to find one that jsdelivr would serve (other recent versions run into the issue where they exceed 100MB); it looks like `https://cdn.jsdelivr.net/npm/ink-mde@0.18.1/+esm` is the most recent working version that is also small enough for jsdelivr to serve up.

**Comment by voracious** [comment]
Can you share a minimal reproduction for this?

**Comment by davidcelis** [comment]
@voracious Sure; I'm including Ink in a Rails application using their `importmap` map feature, but I don't think I'm doing anything super out of the ordinary.

Thankfully, I was able to reproduce this in a bare Rails application with minimal configuration of Ink. Here's a repository: https://github.com/davidcelis/ink-demo

Please let me know if the README isn't enough to get set up; not sure what your experience or comfort level is like with Ruby and Rails, but I'll be happy to help as much as needed!

**Comment by voracious** [comment]
@davidcelis I started looking into this, and it looks like multiple things are broken. It looks like some things are also broken in `0.18.1`, and I'm thinking this has something to do with either import maps or jsdelivr. If you get a chance, could you try to create a reproduction using the jsdelivr CDN on something like CodeSandbox or CodePen? Otherwise, I'll try to do that when I can get back to this.

**Comment by davidcelis** [comment]
Oh Codepen was a much better idea 😅 Here's one that reproduces the issue using jsdelivr absent of Rails: https://codepen.io/davidcelis/pen/YzOqdoQ

I have three import statements in there; the first one is what I've been using, which is jsdeliver's suggested ESM import link. The commented out import statements just target `/dist/index.js` since that's the file that ink's package.json declares as the main file. Both the jsdelivr and unpkg imports fail with this error:

```
Uncaught TypeError: Failed to resolve module specifier "solid-js/web". Relative references must start with either "/", "./", or "../". 
```

**Comment by voracious** [comment]
This is definitely an odd issue. Things seem to work fine when Ink is installed via a bundler (e.g. Vite), so I'm still thinking this has something to do with how jsdelivr is serving it.
https://stackblitz.com/edit/vitejs-vite-qzye6e?file=src%2Fmain.ts,src%2Fstyle.css,index.html&terminal=dev

**Comment by davidcelis** [comment]
Does that mean the only current way to install/include Ink is by including it in a bundle/package? I've been successful so far in avoiding having to add a javascript transpiler or webpacker or node runtime to my Rails app; I know Ink is fairly big but I'm hoping to keep a lighter weight way to include it 😅

**Comment by voracious** [comment]
I would love for it to work correctly over a CDN, so I'm still looking into it. As far as I can tell, this is an upstream issue with CodeMirror and it might have to do with the automatic dependency resolution. Here is an example that shows CodeMirror itself is not tokenizing the Markdown when loaded over the CDN but is working correctly when loaded via a bundler.
https://stackblitz.com/edit/vitejs-vite-ne8f1e?file=src%2Fmain.ts&terminal=dev

I've opened a discussion about it though, so we'll see where it goes.
https://discuss.codemirror.net/t/the-markdown-parser-does-not-seem-to-work-when-codemirror-is-loaded-via-a-cdn/5824

**Comment by davidcelis** [comment]
The maintainer's response didn't give me high hopes, but I actually have a bit of good news; I did a lot more digging and found one of the maintainer's Twitter posts from last year where he's lamenting basically this problem: https://twitter.com/MarijnJH/status/1542098023282515968

In that thread someone mentioned a CDN I hadn't used or heard of before; I swapped out my import to use https://esm.sh/ instead:

```js
import { ink, defineOptions} from 'https://esm.sh/ink-mde@0.20.1?bundle'

// …
```

and it just works! It took several seconds to build the bundled file, but my understanding is that it's cached even on their end, so this ended up very quickly getting me a single file for Ink.

I think that I definitely had just run into a limitation in the average javascript CDN, so I'm not sure how worthwhile it will be for you to keep this issue open. It's clear there's no problem with Ink itself, and since it _seems_ to work correctly with esm.sh, I think recommending them as a CDN for users who want to use a CDN would make sense to me
